js/views/pageNavVw.js

```diff
@@ -368,16 +371,171 @@ module.exports = Backbone.View.extend({
     var self = this;
     var reader = new FileReader();
     reader.onload = function (e) {
+
       self.$el.find('.js-avatarPreview').css('background', 'url(' + e.target.result + ') 50% 50% / cover no-repeat');
-      //self.model.set('tempAvatar', e.target.result);
+      self.model.set('tempAvatar', e.target.result);
+
       //TODO: add canvas resizing here
     };
     reader.readAsDataURL($(e.target)[0].files[0]);
   },
 
   settingsDone: function(e){
+
     "use strict";
+
+
     this.model.set('beenSet',true);
+    var self = this;
+    var server = this.model.get('serverUrl');
+    var profileFormData = new FormData();
+    var settingsFormData = new FormData();
+    var uploadImageFormData = new FormData();
+
+
+    if($('textarea#aboutInput').val() != ""){
+        self.model.set('about', $('textarea#aboutInput').val());
+    }
+
+    if($('#storeHandleInput').val() != "" && /^@/.test($('#storeHandleInput').val()) ){
+        self.model.set('handle', $('#storeHandleInput').val());
+    }
+
+    if($('#storeNameInput').val() != ""){
+        self.model.set('name', $('#storeNameInput').val());
+    }else if (self.model.get('name') == undefined){
+        //otherwise error since the profile api needs the name parameter and as of now it is not set in the userMd.js
+        self.model.set('name', "");
+    }
+
+    var themeRadioButtons = $('input[name=theme-selection]');
+    var themeSelected =   ($(themeRadioButtons.filter(':checked'))[0] !=undefined);
+    if(themeSelected){
+       var themeId = $(themeRadioButtons.filter(':checked')[0]).attr('id');
```

Comment:
Could you consolidate this by checking if themeId had a value, rather than first checking if themeSelected is true? If no themes are selected, then themeId should be false.